hearth/spec/hearth/interceptor_spec.rb

```diff
@@ -28,34 +28,22 @@ module Hearth
           .not_to raise_error
       end
 
-      describe 'singleton methods' do
-        Interceptor.hooks.each do |hook|
-          it "defines a singleton method for #{hook}" do
-            expect(subject).to respond_to(hook)
-          end
-
-          it "calls the callback for #{hook}" do
-            expect(callable).to receive(:call).with(context)
-            subject.send(hook, context)
-          end
-        end
-      end
-
       it 'ignores unknown hooks' do
         interceptor = Interceptor.new(unknown: callable)
         expect(interceptor).not_to respond_to(:unknown)
       end
     end
 
-    describe '#callbacks' do
-      it 'returns a hash of callbacks' do
-        expect(subject.callbacks).to be_a(Hash)
-        expect(subject.callbacks.keys).to include(*Interceptor.hooks)
-      end
+    describe 'singleton methods' do
+      Interceptor.hooks.each do |hook|
+        it "defines a singleton method for #{hook}" do
+          expect(subject).to respond_to(hook)
+        end
 
-      it 'ignores unknown hooks' do
-        interceptor = Interceptor.new(unknown: callable)
-        expect(interceptor.callbacks).not_to have_key(:unknown)
+        it "calls the callback for #{hook}" do
+          output = subject.send(hook, context)
+          expect(output).to be context
```

Comment:
This seems like it could happen in a few ways - maybe should assert the callback actually receives call

Reply:
I had it that way before, but I changed it when trying to change the define Singleton method approach. I can change this back.